### PR TITLE
Add gar_wg_state and gar_bp_state variables to Sync.py

### PR DESCRIPTION
**Pull request by AlaskaJedi**
Error accessing variable 'gar_wg_state' when only using Withings blood pressure device.  Moved set_lastsyn() under reach fit_data if statements so the other variable isn't used..

UnboundLocalError: cannot access local variable 'gar_wg_state' where it is not associated with a value

### CODE REVIEW

**Comment by stynoo** [comment]
Oh, I see what's going on here.  This most likely also explains the duplicate entries we see sometimes in the blood pressure stats.  
There is no need to duplicate the code block, just declare both variables at the start of the function.
Try something like this:
```
        # Upload to Garmin Connect
        if ARGS.garmin_username and (
            fit_data_weight is not None or fit_data_blood_pressure is not None
        ):
            logging.debug("attempting to upload fit file...")
            gar_wg_state = None
            gar_bp_state = None
```

**Comment by AlaskaJedi** [comment]
@stynoo thanks, that is cleaner than my approach, and I can confirm it works when syncing only BP data. (I don't own the scale)

File: withings_sync/sync.py [hunk 1]
```diff
@@ -470,6 +472,7 @@ def sync():
                     logging.info(
                         "Fit file with blood pressure information uploaded to Garmin Connect"
                     )
+                    # Save this sync so we don't re-download the same data again (if no range has been specified)
```
Comment (stynoo) [inline withings_sync/sync.py:475]: There is no longer need for this comment here.

Comment (AlaskaJedi) [reply withings_sync/sync.py]: removed, thanks!

**Comment by stynoo** [comment]
lgtm!

**Pull request merged**